Mathlib/Analysis/Complex/UpperHalfPlane/Basic.lean

```diff
@@ -163,6 +163,16 @@ theorem normSq_ne_zero (z : ℍ) : Complex.normSq (z : ℂ) ≠ 0 :=
 theorem im_inv_neg_coe_pos (z : ℍ) : 0 < (-z : ℂ)⁻¹.im := by
   simpa using div_pos z.property (normSq_pos z)
 
+lemma ne_nat (z : ℍ) : ∀ n : ℕ, z.1 ≠ n := by
+  intro n
+  have h1 := z.2
+  aesop
+
+lemma ne_int (z : ℍ) : ∀ n : ℤ, z.1 ≠ n := by
+  intro n
```

Comment:
The proof of `UpperHalfPlane.coe_mem_integerComplement` mildly suggests that these lemmas should be turned around, but I'm not going to push for this.

Reply:
Hmm sure, I'll see if it becomes annoying later on with the use case I have for this, if so I'll come back and change them.